pkg/networkserver/grpc_gsns.go

```diff
@@ -287,7 +292,12 @@ func (ns *NetworkServer) matchAndHandleDataUplink(ctx context.Context, dev *ttnp
 				}
 				ctx = log.NewContextWithField(ctx, "f_cnt_reset", true)
 
-				macState, err := mac.NewState(dev, ns.FrequencyPlans, ns.defaultMACSettings)
+				fps, err := ns.FrequencyPlansStore(ctx)
```

Comment:
You're in the same function (`matchAndHandleDataUplink`), do you need to refresh this here ?